projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -43,13 +43,15 @@ export class AuthService {
    * Check params in url and if there is an code/token then try to login with those.
    */
   async checkOAuthParamsInUrl(): Promise<void> {
-    try {
-      const tokenReceived = getLastValueSync(
-        this.oAuthLibWrapperService.events$.pipe(
-          filter((e) => e.type === 'token_received')
-        )
-      );
+    let tokenReceived: OAuthEvent | undefined;
+    const sub: Subscription = this.oAuthLibWrapperService.events$
+      .pipe(
+        take(1),
+        filter((e: OAuthEvent) => e.type === 'token_received')
```

Comment:
As far as I understand: In case when other unrelated event is emitted first from events$ (due to any reason), we stop listening immediately and miss the event `token_received` that is emitted later. 

To avoid this issue, I suggest to filter() first, and only later take(1). 
```suggestion
        filter((e: OAuthEvent) => e.type === 'token_received')
        take(1),
```